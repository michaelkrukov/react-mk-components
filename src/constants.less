// Colors
@primary: #5F1EC4;
@secondary: #E0E0E0;
@danger: #dc3545;
@success: #28a745;
@warning: #f8ac19;
@info: #17a2b8;
@dark: #343a40;
@light: #f8f9fa;

// Sizes
@sm: 576px;
@md: 768px;
@lg: 992px;
@xl: 1200px;

@c-sm: 540px;
@c-md: 720px;
@c-lg: 960px;
@c-xl: 1140px;
