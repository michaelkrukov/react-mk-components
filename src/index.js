export {
    default as Button
} from './components/Button';
export {
    default as Container
} from './components/Container';
export {
    default as Modal,
    ModalHeader,
    ModalBody,
    ModalFooter
} from './components/Modal';
export {
    default as Transition
} from './components/Transition';
export {
    default as BodyEnd
} from './components/BodyEnd';
