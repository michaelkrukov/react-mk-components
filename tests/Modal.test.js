import React from 'react';
import { Modal } from '../src';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';


const ModalComponent = (props) => {
  const [hidden, setHidden] = useState(true);

  const toggle = () => setHidden(!hidden);

  return (
    <div>
      <Button info block onClick={toggle}>Toggle</Button>
      <Modal hidden={hidden} toggle={toggle} {...props}>
        <div style={{padding: '1rem', textAlign: 'center'}}>
          Hello there!<br/>
          Hello there!<br/>
          Hello there!<br/>
          Hello there!<br/>
        </div>
      </Modal>
    </div>
  );
};

test('Modal renders', () => {
  const component = renderer.create(
    <ModalComponent />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Modal changes hidden', () => {
  const wrapper = mount(<ModalComponent />);

  expect(wrapper.find(Modal).at(0).prop('hidden')).toEqual(true);

  wrapper.find('button').at(0).simulate('click');

  expect(wrapper.find(Modal).at(0).prop('hidden')).toEqual(false);
});
