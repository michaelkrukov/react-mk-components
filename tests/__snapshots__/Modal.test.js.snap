// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
<div>
  <button
    className="Button info block"
    onClick={[Function]}
    type="button"
  >
    Toggle
  </button>
  <div
    className="TransitionFade exit quick"
  >
    <div
      className="ModalWrapper"
      onClick={[Function]}
      onMouseDown={[Function]}
      role="region"
    >
      <div
        className="ModalDialog centered"
      >
        <div
          className="ModalContent"
        >
          <div
            style={
              Object {
                "padding": "1rem",
                "textAlign": "center",
              }
            }
          >
            Hello there!
            <br />
            Hello there!
            <br />
            Hello there!
            <br />
            Hello there!
            <br />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ModalBackdropComponent static"
    />
  </div>
</div>
`;
